ui/src/parts/lightbox.tsx

```diff
@@ -0,0 +1,224 @@
+    lazyImageObserver = new window.IntersectionObserver(entries =>
+      entries.forEach(entry => {
+        if (entry.isIntersecting) {
+          const lazyImage = entry.target as HTMLImageElement
+          lazyImage.src = lazyImage.dataset.src!
+          lazyImage.classList.remove("lazy")
+          lazyImageObserver.unobserve(lazyImage)
+        }
+      })
+    ),
+    handleKeyDown = (ev: KeyboardEvent) => {
+      if (['Escape', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].indexOf(ev.key) >= 0) ev.preventDefault()
+      if (ev.key === 'Escape') lightboxB(null)
+      if (ev.key === 'ArrowRight') setActiveImageIdx(prevIdx => prevIdx === images.length - 1 ? 0 : prevIdx + 1)
+      else if (ev.key === 'ArrowLeft') setActiveImageIdx(prevIdx => prevIdx === 0 ? images.length - 1 : prevIdx - 1)
+    }
+
+  React.useEffect(() => {
+    // Initialize intersection observer for lazy images.
+    const lazyImages = [].slice.call(document.querySelectorAll(".lazy"))
+    lazyImages.forEach(lazyImage => lazyImageObserver.observe(lazyImage))
+
+    // Add keyboard events listener.
+    window.addEventListener("keydown", handleKeyDown)
+    return () => window.removeEventListener("keydown", handleKeyDown)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [])
+
+  // Handle image navigation scroll.
+  React.useEffect(() => {
+    if (isGallery && imageNavRef.current) {
+      const
+        navRef = imageNavRef.current,
+        scrollLeftMiddle = (navRef.clientWidth / 2) - 62,
+        scrollLeftActiveImage = activeImageIdx * 124,
+        scrollBehavior = defaultScrollSetRef.current ? 'smooth' : 'auto'
+      if (activeImageIdx === 0) navRef.scrollLeft = 0
+      else if (activeImageIdx === images.length - 1) navRef.scrollLeft = navRef.scrollWidth - navRef?.clientWidth
+      else if (scrollLeftActiveImage > scrollLeftMiddle) navRef.scrollTo({ left: scrollLeftActiveImage - scrollLeftMiddle, behavior: scrollBehavior })
+      else navRef.scrollTo({ left: 0, behavior: scrollBehavior })
+
+      if (!defaultScrollSetRef.current) defaultScrollSetRef.current = true
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [activeImageIdx])
+
+  return (
+    <div className={css.body}>
+      <div className={css.header}>
+        <Fluent.ActionButton
+          styles={iconStyles}
+          onClick={() => lightboxB(null)}
+          iconProps={{ iconName: 'Cancel', style: styles.icon }}
+        />
+      </div>
+      <div className={css.content} style={{ height: `calc(100% - ${HEADER_HEIGHT + FOOTER_HEIGHT}px)` }}>
+        <img className={css.img} alt={title} src={getImageSrc(images[activeImageIdx])} />
+        {isGallery &&
+          <>
+            <Fluent.ActionButton
+              styles={iconStyles}
+              className={css.iconStylesRootArrow}
+              style={{ left: 0 }}
+              onClick={() => setActiveImageIdx(activeImageIdx === 0 ? images.length - 1 : activeImageIdx - 1)}
```

Comment:
Since next state for `activeImageIdx` depends on the previous one, let's use the value provided by the setState callback rather than the closure one as we discussed. Same for the other button.